crates/orchestrator/src/lib.rs

```diff
@@ -416,11 +436,29 @@ where
     }
 
     // Assumes nodes do not post 'ready' twice
-    // TODO ED Add a map to verify which nodes have posted they're ready
-    fn post_ready(&mut self) -> Result<(), ServerError> {
+    fn post_ready(&mut self, pubkey: &KEY) -> Result<(), ServerError> {
+        // If we have not disabled registration verification.
+        if !self.config.disable_registration_verification {
+            // Is this node allowed to connect?
+            if !self.config.public_keys.contains(pubkey) {
```

Comment:
I think we can replace both of these and just check `known_nodes_with_stake.contains(pubkey)` (a node has to register before posting ready)

Reply:
This one was painful 😅 but we got it.